quesma/frontend_connectors/vitess_mysql_connector.go

```diff
@@ -0,0 +1,166 @@
+// Copyright Quesma, licensed under the Elastic License 2.0.
+// SPDX-License-Identifier: Elastic-2.0
+
+// Experimental alpha processor for MySQL protocol
+
+package frontend_connectors
+
+import (
+	"context"
+	"errors"
+	"time"
+	"vitess.io/vitess/go/mysql"
+	"vitess.io/vitess/go/mysql/replication"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/vt/proto/query"
+	"vitess.io/vitess/go/vt/vtenv"
+
+	"github.com/QuesmaOrg/quesma/quesma/logger"
+	"github.com/QuesmaOrg/quesma/quesma/quesma/recovery"
+	quesma_api "github.com/QuesmaOrg/quesma/quesma/v2/core"
+)
+
+type VitessMySqlConnector struct {
+	processors []quesma_api.Processor
+	listener   *mysql.Listener
+	endpoint   string
+}
+
+func NewVitessMySqlConnector(endpoint string) (*VitessMySqlConnector, error) {
+	connector := VitessMySqlConnector{
+		endpoint: endpoint,
+	}
+
+	// FIXME: the parameter values below should be tweaked, in particular (list not exhaustive):
+	// - timeouts, delays are set to time.Second * 0
+	// - authServer is set to mysql.NewAuthServerNone(), meaning no authentication
+	// - TLS is not set up
+	listener, err := mysql.NewListener("tcp", endpoint, mysql.NewAuthServerNone(), &connector, time.Second*0, time.Second*0, false, false, time.Second*0, time.Second*0)
+	if err != nil {
+		return nil, err
+	}
+	connector.listener = listener
+
+	return &connector, nil
+}
+
+func (t *VitessMySqlConnector) Listen() error {
+	go func() {
+		defer recovery.LogPanic()
+		t.listener.Accept()
+	}()
+	return nil
+}
+
+// Implementation of Vitess mysql.Handler interface:
+
+type ComQueryMessage struct {
+	Conn  *mysql.Conn
```

Comment:
Seems unused for now.  This makes `frontend_connector` and `processor` strongly coupled. I would suggest to remove it for now

Reply:
Removed it in https://github.com/QuesmaOrg/quesma/pull/1212/commits/4d23a21ec3811df6e283e717e4f67b633213f446